### PR TITLE
Update some out-of-date information

### LINKED ISSUE #1040: Suggestions to use system LLVM link to a section that no longer exists
At the bottom of [Getting Started - Configuring the compiler](https://rustc-dev-guide.rust-lang.org/getting-started.html#configuring-the-compiler) there is a sentence about using the system LLVM that includes a link that would have previously had instructions on how to set that up.
> You may also want to set up [system LLVM](https://rustc-dev-guide.rust-lang.org/building/suggested.html#skipping-llvm-build) to avoid building LLVM from source.

<br>

Additionally, [Adding a new target - Specifying a new LLVM](https://rustc-dev-guide.rust-lang.org/building/new-target.html#specifying-a-new-llvm) contains a link to the same page, suggesting
> If you have a local LLVM checkout that is already built, you may be able to configure Rust to treat your build as the [system LLVM](https://rustc-dev-guide.rust-lang.org/building/suggested.html#skipping-llvm-build) to avoid redundant builds.

<br>

The section linked was removed in #1030 and the link now simply points to the top of the page for Suggested Workflows.  From the content of #1030 I believe that the suggestion to use the system LLVM should be removed entirely in favor of downloading CI LLVM.

### PR DESCRIPTION
**Pull request by camelid**
Fixes #1040. (And some other stuff.)

### CODE REVIEW

File: src/appendix/humorust.md [hunk 1]
```diff
@@ -10,4 +10,4 @@ enlightening?
 - [Rust Koans](https://users.rust-lang.org/t/rust-koans/2408)
 - [`break rust;`](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=0ab2bd6a9d722e0f05a95e2a5dcf89cc)
 - [The Nomicon Intro](https://doc.rust-lang.org/stable/nomicon/)
-- [`rustc-ty` renaming punfest](https://rust-lang.zulipchat.com/#narrow/stream/131828-t-compiler/topic/rustc-ty.20naming.20bikeshed.20.2F.20punfest.20(was.3A.20design.20meeting.202.2E.2E.2E/near/189906455 )
+- [`rustc-ty` renaming punfest](https://rust-lang.zulipchat.com/#narrow/stream/131828-t-compiler/topic/rustc-ty.20naming.20bikeshed.20.2F.20punfest.20\(was.3A.20design.20meeting.202.2E.2E.2E/near/189906455)
```
Comment (camelid) [inline src/appendix/humorust.md:13]: This link was broken because the link had an unescaped parenthesis.

File: src/backend/updating-llvm.md [hunk 2]
```diff
@@ -66,9 +66,9 @@ Example PRs look like:
 
 ## Feature updates
 
-> Note that this is all information as applies to the current day in age. This
-> process for updating LLVM changes with practically all LLVM updates, so this
-> may be out of date!
+> Note that this information is as of the time of this writing (December 2018
+> <!-- date: 2018-12 -->. The process for updating LLVM changes with practically
```
Comment (camelid) [inline src/backend/updating-llvm.md:70]: The date-checker hasn't landed yet, but I think it will soon and this doesn't cause any harm, so I thought it was worth adding a date-check comment.

File: src/building/new-target.md [hunk 3]
```diff
@@ -8,28 +8,22 @@ relevant to your desired goal.
...
 cd ..
 git add llvm_target
 git commit -m 'Use my custom LLVM'
 ```
 
-If you have a local LLVM checkout that is already built, you *may* be
-able to configure Rust to treat your build as the [system
-LLVM][sysllvm] to avoid redundant builds.
-
-[sysllvm]: ./suggested.md#skipping-llvm-build
```
Comment (camelid) [inline src/building/new-target.md:31]: It seems unfortunate to have to delete this information, but the key information was already deleted.

File: src/traits/chalk.md [hunk 4]
```diff
@@ -35,7 +35,7 @@ and designs for the trait system.
 
 **rustc**. Once we are happy with the logical rules, we proceed to
 implementing them in rustc. We map our struct, trait, and impl declarations
-into logical inference rules in the [lowering module in rustc](./lowering-module.md).
+into logical inference rules in the lowering module in rustc.
```
Comment (camelid) [inline src/traits/chalk.md:38]: I think this file was removed in #1013.

File: src/backend/updating-llvm.md [hunk 5]
```diff
@@ -66,9 +66,9 @@ Example PRs look like:
 
 ## Feature updates
 
-> Note that this is all information as applies to the current day in age. This
-> process for updating LLVM changes with practically all LLVM updates, so this
-> may be out of date!
+> Note that this information is as of the time of this writing (December 2018
```
Comment (jyn514) [inline src/backend/updating-llvm.md:69]:
```suggestion
> Note that this information is as of the time of this writing (December 2018)
```

File: src/building/new-target.md (hunk 3 above)
Comment (jyn514) [reply src/building/new-target.md:31]: I think rather than deleting this, we should move the system-llvm section here. I hadn't thought of someone who had a custom LLVM fork already built, this would be useful for them.

**Comment by camelid** [comment]
I've wanted to see info about metadata and rustlibs in the dev-guide for a
while, so thank you!

**Comment by jyn514** [comment]
I think you commented on the wrong PR.

**Comment by camelid** [comment]
Yes, I was just about to say that 😆

**Comment by jyn514** [comment]
ping @camelid - will you have time to look at this soon?

**Comment by camelid** [comment]
Hopefully, yes.

File: src/backend/updating-llvm.md (hunk 2 above)
Comment (camelid) [reply src/backend/updating-llvm.md:70]: It has landed!

File: src/getting-started.md [hunk 6]
```diff
@@ -150,11 +150,15 @@ To get started, try one of the following commands:
 For more suggestions, see https://rustc-dev-guide.rust-lang.org/building/suggested.html
 ```
 
-You may also want to set up [system LLVM][sysllvm] to avoid building LLVM from source.
+Note that by default, `x.py setup` will use CI-built LLVM if available for your
+platform so that you don't need to build LLVM in addition to building the
+compiler. In some circumstances, such as when updating the version of LLVM used
+by `rustc`, you may want to temporarily disable this feature. See the ["Updating
+LLVM"] for more.
```
Comment (camelid) [inline src/getting-started.md:157]:
```suggestion
LLVM"] section for more.
```

File: src/building/new-target.md (hunk 3 above)
Comment (camelid) [reply src/building/new-target.md:31]: What is the "system-llvm" section? Do you mean the part I added the begins with

> Note that by default, `x.py setup` will use CI-built LLVM

?

Comment (jyn514) [reply src/building/new-target.md:31]: > What is the "system-llvm" section?

I mean the section I removed in https://github.com/rust-lang/rustc-dev-guide/pull/1030.

Comment (camelid) [reply src/building/new-target.md]: Okay, does this look good: 2d42cf7?

Comment (jyn514) [reply src/building/new-target.md]: Looks great, thanks!

File: src/getting-started.md [hunk 7]
```diff
@@ -123,17 +123,17 @@ The compiler has a configuration file which contains a ton of settings. We will
...
 
-```sh
+```console
 $ x.py setup
 ```
 
 This will walk you through an interactive setup for x.py that looks like this:
 
-```
+```console
```
Comment (jyn514) [inline src/getting-started.md:136]: Can you please avoid these unrelated changes here? See also https://github.com/rust-lang/rustc-dev-guide/issues/515.

File: src/getting-started.md [hunk 8]
```diff
@@ -150,11 +150,15 @@ To get started, try one of the following commands:
 For more suggestions, see https://rustc-dev-guide.rust-lang.org/building/suggested.html
 ```
 
-You may also want to set up [system LLVM][sysllvm] to avoid building LLVM from source.
+Note that by default, `x.py setup` will use CI-built LLVM if available for your
+platform so that you don't need to build LLVM in addition to building the
+compiler. In some circumstances, such as when updating the version of LLVM used
+by `rustc`, you may want to temporarily disable this feature. See the ["Updating
+LLVM" section] for more.
```
Comment (jyn514) [inline src/getting-started.md:158]:
````suggestion
If you want to download LLVM without running `x.py setup`, you can set
```toml
[llvm]
download-ci-llvm = true
```
````

**Review by jyn514 — CHANGES_REQUESTED** [review]

**Comment by spastorino** [comment]
This looks good, I think after addressing @jyn514 comments is good to go.

File: src/getting-started.md (hunk 7 above)
Comment (camelid) [reply src/getting-started.md]: Heh, I don't even remember making these changes 😆

**Review by jyn514 — APPROVED** [review]

**Comment by piegamesde** [comment]
I added

```
[target.x86_64-unknown-linux-gnu]
llvm-config = "/path/to/llvm/llvm-7.0.1/bin/llvm-config"
```

to my `./.cargo/config` and I then get the following error: "error: expected a table, but found a string for `target.x86_64-unknown-linux-gnu.llvm-config`"
